src/test/codegen/issue-75525-bounds-checks.rs

```diff
@@ -0,0 +1,27 @@
+// Regression test for #75525, verifies that no bounds checks are generated.
+
+// min-llvm-version: 12.0.0
+// compile-flags: -O
+
+#![crate_type = "lib"]
+
+// CHECK-LABEL: @f0
+// CHECK-NOT: panic
+#[no_mangle]
+pub fn f0(idx: usize, buf: &[u8; 10]) -> u8 {
+    if idx < 8 { buf[idx + 1] } else { 0 }
```

Comment:
NB: This test case does not appear to have generated a bounds check before 12.0.